### DISCUSSION TITLE
[Detector Support]: Just got OpenVino capable N100.  Any reason to keep using Coral?  OpenVino vs Coral?

### DISCUSSION THREAD

**Discussion by rhatguy** [blakeblackshear/frigate#11717] [Detector Support]
I recently purchased an N100 mini PC from Amazon.  The N100 has Intel UHD 730 GPU.  I've set frigate up running inside docker on Ubuntu 22.04 along with using OpenVino for detection.  I'm running the newest Frigate .14 beta and the detector CPU graph shows usage usually less than 12% (I'm guessing this is averaged and there are probably spikes higher than that), and total CPU utilization hovers around 25%-30% (with 6 * 4k cameras with 720p detect streams).  Initial impressions seem like OV is working great.

I also have a Coral that I previously used with a Raspberry PI 4 that I have not plugged into the N100 yet.

My question is, is there any advantage of using the Coral as opposed to the N100 GPU for detection?  I understand the Coral is very efficient and would take workload off of the CPU, but this machine will be solely used for Frigate so as long as it works, I don't really care how loaded the CPU is.  Is the Coral any better/worse at detection?  Any other reasons to use Coral over OpenVino that I'm missing?

### Version

0.14.0-bccffe6

### Operating system

Debian

### Install method

Docker Compose

### Coral version

CPU (no coral)

**Comment by NickM-27** [comment]
you might have a tad lower power usage with the coral. Quality of detection comes down to the model that is run not the hardware itself

**Reply by HonkStonk** [reply]
I have the same and more questions and similar situation. 

Compared to the Intro page that indicates 5-10ms inference speed with N100+Coral - how does that compare to N100+OpenVino?

I have an ”unused” N200 (Asus PN42) that I want to run Frigate on (along sporadic Plex use) and now it looks like Ubuntu 24.04 is supported by OpenVino. 

What I also discovered was that my mini PC can take an inexpensive Coral A+E key module instead of my unused WiFi module. Don’t dare to take a chance on the double coral with E key since it doesn’t seem guaranteed the WiFi M.2 port has two separate lanes.

So I wonder:
Is N200 with latest OpenVino much better than N100? Guess so.

Will a M.2 PCIe Coral lower my N200 PC power consumption notably in the long run? Now it consumes 4-5W during very light loads like my Pi normally.

**Reply by rhatguy** [reply]
I have an N100 and a coral and have tried both it and OpenVino.  I believe you’ll see lower power utilization with the coral as it’s more efficient.  Whether or not it’s enough to matter is another question.  I wouldn’t spend extra money on a new coral if I didn’t have one, but since I already had one I chose to use it.

**Reply by NickM-27** [reply]
Openvino on n100 with the default model is ~15ms or so, larger models like yolonas can be fun but inference time will be higher ~25ms

> and now it looks like Ubuntu 24.04 is supported by OpenVino.

To be clear that's not relevant since frigate runs in docker, as long as the driver is installed on the host.

**Reply by tve** [reply]
> Quality of detection comes down to the model that is run not the hardware itself

Assuming all the models are 8-bit quantized and are trained on the same image sizes, etc. Is that the case?

**Reply by stefajo** [reply]
I'm very new to this but I'd like people to know my results using a Ryzen 3700x with openvino. Around 10ms running openvino cpu and hardware acceleration with my 6700xt gpu. Pretty sure I can optimize this because 500% cpu usage on detection doesn't make sense. Unfortunately this is through Home Assistant so I don't think I can get it to run the AMD rocm version...
![Screenshot_20241225_185315_Home Assistant](https://github.com/user-attachments/assets/a5ab6ad1-b9b6-4604-8885-dea329f59a95)
![Screenshot_20241225_185328_Home Assistant](https://github.com/user-attachments/assets/f41cc5e7-b3e8-4aab-913c-db14fdc61903)

**Reply by NickM-27** [reply]
500% CPU on detection does make sense, given that it is running on the CPU. That is 100% of 5 CPU cores

**Reply by stefajo** [reply]
That's interesting. I'm not finding any documentation and adjusting the amount of cores used. I assume it's using the max amount available. Would options like these and using my AMD gpu be available on standalone builds of Frigate? Home Assistant seems to limit certain capabilities.

**Reply by NickM-27** [reply]
You can control cpu detectors in your config. And yes using ROCm requires running in docker directly

**Reply by stefajo** [reply]
My yaml config for frigate detectors 

detectors:
  ov:
    type: openvino
    device: CPU

I'm not seeing anything in the official Frigate or Openvino documentation on further adjusting CPU parameters when using ov.

**Reply by NickM-27** [reply]
Ah right, yeah using opencv there is no direct control in the frigate config, although some env variables seem to exist https://blog.openvino.ai/blog-posts/cpu-dispatcher-control-for-openvino-tm-inference-runtime-execution

**Reply by stefajo** [reply]
Thank you, this is exactly what I need to learn more about this. I will likely set separate Home Assistant and Frigate so that I can run them side by side in Proxmox.

**Reply by stefajo** [reply]
I see you are a community developer for Frigate and the Home Assistant integration. I will report back after several days with CPU detection inference speeds comparison. I don't see too many AMD posts, so I'd like to contribute what data i can.

**Reply by HonkStonk** [reply]
FYI - my N200+OpenVino (GPU) now does 11-12ms inference speed always on 2 tapo cameras (one 1080 and one 1440) - integrates great with my HA on RPI5 as soon as I found the cards themselves had a lot of config options. 

The Asus PN42 motherboard seemed impossible to configure for E-key coral instead of WiFi so I gave that up. But with my 11ms with no additional hardware I think I’m good.

The only thing that is a performance related issue now is that the setup is not quick enough to detect persons leaving the house - only entering…

**Reply by NickM-27** [reply]
> The only thing that is a performance related issue now is that the setup is not quick enough to detect persons leaving the house - only entering…

are you sure that's the issue, or can you elaborate? if inference time is 11ms then unless you have more than 10 cameras it shouldn't be a matter of speed

**Reply by HonkStonk** [reply]
Well - I’m not sure where the ”bottleneck” in the system is but for sure you have to stand in front of the camera for 4 consecutive seconds before it logs and reports a person as detected or it will discard it altogether. I leave the house quicker than that but enter the house longer than that. 

I should note the camera points down from porch ceiling and sees only heads from above. Very good accuracy when entering at least.

Does it have to do with the models (standard mobilenetssd) or with N200/OpenVino or with the mqtt connection to my Pi HA? I would expect the Frigate app in my standalone Ubuntu pc would record anything with motion and post process anything in due time and then report to HA it detected a person but from all my testing it does not report anything unless I stand in front of it for 4-5 seconds. Is there any filter to tune?

**Reply by NickM-27** [reply]
The default models trained on the COCO dataset have a difficult time with top-down camera angle due to no images like that being in the dataset. The easiest way to know what is going on is to see a recording of the debug live view during this scenario

**Reply by Zapnologica** [reply]
Anyfeed back on this, i want to ditch coral tpu to use n100 openvino on IGPU

This will also give us better models not?

**Reply by hawkeye217** [reply]
Yes, an iGPU can run much more accurate models than the Coral.

**Comment by bbrodriguez** [comment]
I dumped my coral and an accepted the higher inference times of the GPU to use YOLOv9 (N95) and never looked back.
